src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/LinkDesignerBase.cs

```diff
@@ -60,6 +60,39 @@ protected bool FindResourceDesigner (AssemblyDefinition assembly, bool mainAppli
 
 				}
 			}
+
+			if (string.IsNullOrEmpty(designerFullName)) {
+				LogMessage ($"Inspecting member references for assembly: {assembly.FullName};");
+				var memberRefs = assembly.MainModule.GetMemberReferences ();
+				var memberIdx = 0;
+				foreach (var memberRef in memberRefs) {
+					memberIdx++;
+					string declaringType = memberRef.DeclaringType?.ToString () ?? string.Empty;
+					if (!declaringType.Contains (".Resource/")) {
+						continue;
+					}
+					if (declaringType.Contains ("_Microsoft.Android.Resource.Designer")) {
+						continue;
+					}
+					var resolved = false;
+					try {
+						var def = memberRef.Resolve ();
+						if (resolved = def != null) {
+							LogMessage ($"Resolved member `{memberRef?.Name}`");
+						}
+					} catch (Exception ex) {
+						LogMessage ($"Exception resolving member `{memberRef?.Name}`: {ex}");
+						resolved = false;
+					}
+					if (!resolved) {
+						LogMessage ($"Adding _Linker.Generated.Resource to {assembly.Name.Name}");
```

Comment:
@jonathanpeppers, @dellis1972: do we want to *also* record *why* we're adding `_Linker.Generated.Resource`?  For example, because `memberRef.Name` with type `memberRef.DeclaringType` could not be resolved?  Or would this just be extraneous information?

Reply:
The downside with that suggestion is that `LogMessage()` messages are *hidden by default*, unless you build with `-p:_ExtraTrimmerArgs=--verbose`.